src/Simplex_tree/include/gudhi/Simplex_tree/multi_filtrations/Finitely_critical_filtrations.h

```diff
@@ -0,0 +1,1051 @@
+        return stuff;
+      if (stuff.is_inf())
+        continue;
+      for (std::size_t i = 0; i < stuff.num_parameters(); ++i) {
+        result[i] = std::min(result[i], stuff[i]);
+      }
+    }
+    return result;
+  }
+  /*
+   * returns the smallest value for the poset order that is bigger than all of the values
+   * in this multi filtration
+   */
+  inline OneCritical factorize_above() const {
+    if (this->num_generators() == 0) [[unlikely]]
+      return OneCritical();
+    OneCritical result(multi_filtration_[0].num_parameters(), -OneCritical::T_inf);
+    for (auto &stuff : this->multi_filtration_) {
+      if (stuff.is_nan() || stuff.is_inf())
+        return stuff;
+      if (stuff.is_minus_inf())
+        continue;
+      for (std::size_t i = 0; i < stuff.num_parameters(); ++i) {
+        result[i] = std::max(result[i], stuff[i]);
+      }
+    }
+    return result;
+  }
+  /** \brief This functions take the filtration value `this` and pushes it to
+   * the cone \f$ \{ y\in \mathbb R^n : y>=x \} \f$. After calling this method,
+   * the value of this is updated to \f$ \mathrm{this} = \min \{ y\in \mathbb
+   * R^n : y>=this \}\cap \{ y\in \mathbb R^n : y>=x \}
+   * @param[in] x The target filtration value on which to push `this`.
+   */
+  inline void push_to(const One_critical_filtration<T> &x) {
+    if (this->is_inf() || this->is_nan() || x.is_nan() || x.is_minus_inf())
+      return;
+    if (x.is_inf() || this->is_minus_inf()) {
+      *this = x;
+      return;
+    }
+    for (auto &stuff : *this) {
+      stuff.push_to(x);
+    }
+  }
+  // TODO : this is not well defined for kcritical <-> kcritical
+
+  /** \brief This functions take the filtration value `this` and pulls it to the
+   * cone \f$ \{ y\in \mathbb R^n : y<=x \} \f$. After calling this method, the
+   * value of this is updated to \f$ \mathrm{this} = \max \{ y\in \mathbb R^n :
+   * y<=this \}\cap \{ y\in \mathbb R^n : y<=x \}
+   * @param[in] x The target filtration value on which to push `this`.
+   */
+  inline void pull_to(const One_critical_filtration<T> &x) {
+    if (x.is_inf() || this->is_nan() || x.is_nan() || this->is_minus_inf())
+      return;
+    if (this->is_inf() || x.is_minus_inf()) {
+      *this = x;
+      return;
+    }
+    for (auto &stuff : *this) {
+      stuff.pull_to(x);
+    }
+  } 
+  // cannot be const, as gudhi iterators are not const
+  template <typename U> inline U linear_projection(const std::vector<U> &x) {
+    if constexpr (co) {
+      U projection = std::numeric_limits<U>::lowest();
+      for (const auto &y : *this) {
+        projection = std::max(projection, y.linear_projection(x));
+      }
+      return projection;
+    } else {
+      U projection = std::numeric_limits<U>::max();
+      for (auto &y : *this) { // cannot be const (Gudhi)
+        projection = std::min(projection, y.linear_projection(x));
+      }
+      return projection;
+    }
+  }
+
+  /*
+  * Checks if b is cleanable with respect to a
```

Comment:
(tested as `a >= b`)
In my interpretation as a union of upper-right quadrants, it is the larger point that can be removed, not the smaller one.